### PR TITLE
Export Fix

### LINKED ISSUE #132: export to vtk incomplete in unbalanced parallel partitions. needed to add MPI barrier in lua for sucessful completion
This could be connected to the general pass-through we wanted to do regarding the pertinence of MPI Barriers dispersed throughout the code.

However, this is an issue that is pretty annoying that I amsingling it out. When export to VTK for a parallel simulations that has severely unbalanced partition, the code may exit before all ranks are done writing their .vtu portion. This can happen when exporting just the mesh or when exporting field functions (including multiple field functions).

I was able to fix this from lua by adding in lua an MPIBarrier() call., example below. We should not rely ont he user to know or do this so I suggest that we add barriers in the various functions that perform write to disk (export to vtk, write field functions, write multiple field functions, maybe the split mesh business as well, and possibly others.

```
MeshHandlerExportMeshToVTK("Stage_3")
MPIBarrier()
```

### PR DESCRIPTION
**Pull request by rodriguez-jay**
Closes #132

### CODE REVIEW

**Comment by andrsd** [comment]
I am going to re-open this. This is correct. I will just edit the PR description...

File: lua/framework/mesh/export/lua_mesh_export.cc
Comment (andrsd) [inline lua/framework/mesh/export/lua_mesh_export.cc:2]: Remove this line and your PR is good to go in as far as I can tell.

**Review by andrsd — CHANGES_REQUESTED** [review]

**Review by ragusa — APPROVED** [review]
do the change requested by David. I am going to approve as his request will still block this PR until resolved.

**Comment by rodriguez-jay** [comment]
Should be good to go

**Comment by andrsd** [comment]
I don't understand what is going on here. I asked you to fix your patch. You closed the PR, deleted your feature branach and said this is good to go in? 😕

**Comment by rodriguez-jay** [comment]
I thought I closed the PR on my personal fork. After merging my branch into main. Did I accident close the PR here?

**Comment by andrsd** [comment]
Well, the issue number is #132. 142 was the PR number. IDK what you did in your personal fork, but it hosed up this PR.

However, I salvaged and fixed your code in PR #145 . You will get credit for the code change, the patch has you as an author...

**Comment by rodriguez-jay** [comment]
Thanks for the help David. Sorry for wrinkling things up there.

**Pull request closed without merging**
